modules/javafx.graphics/src/main/native-prism-sw/JAbstractSurface.c

```diff
@@ -52,9 +52,14 @@ surface_initialize(JNIEnv* env, jobject surfaceHandle) {
 }
 
 JNIEXPORT void JNICALL
-Java_com_sun_pisces_AbstractSurface_nativeFinalize(JNIEnv* env,
-        jobject objectHandle) {
-    disposeNativeImpl(env, objectHandle);
+Java_com_sun_pisces_AbstractSurface_disposeNative(JNIEnv *env, jclass cls, jlong nativePtr)
+{
+    AbstractSurface* surface = (AbstractSurface*) JLongToPointer(nativePtr);
+
+    if (surface != NULL) {
+        surface->cleanup(surface);
+        surface_dispose(&surface->super);
+    }
 }
```

Comment:
there are no checks on memory errors as used to be in the finalize implementation. Are those not needed anymore?

Reply:
Hi Johan,
The `readAndClearMemErrorFlag()` method checks if the variable `mem_Error_Flag` is `JNI_TRUE`.
A call should be made to `setMemErrorFlag()` to set `mem_Error_Flag` to `JNI_TRUE`.

The methods used to dispose Surface and renderer only `free()` the allocated memory and 
do not result in making a call to `setMemErrorFlag().`
So we do not need to check for `readAndClearMemErrorFlag()`
And I think it was not required before this change too.

Reply:
The dispose methods indeed won't set the `mem_Error_Flag` but this flag might have been set by other methods. Isn't there a probability that the flag has been set, and that `readAndClearMemErrorFlag()` is not yet called since the flag has been set?

Reply:
Hi Johan, I did miss to verify this angle.
But have checked the code now and can confirm that, in a given function for all possible calls to `setMemErrorFlag()` there exists a call to `readAndClearMemErrorFlag()` in the same function. So it looks safe to remove the memory checks from `dispose()`

Reply:
I also looked at the code and don't see any mismatches (including those called from the `ACQUIRE_SURFACE` macro). I suppose you could restore the call, but since we are in a disposer thread throwing an exception doesn't seem like the right thing to do. You could log the error, but if we are sure there can be no pending errors, it might not be worth the effort.

Reply:
Yes Kevin, After looking at the code it can be concluded that there are no mismatches. So It seems good to remove those checks. But just in case of safety can keep those as logs too.

Reply:
I don't have a strong preference. Either removing the checks or leaving them in (with a log message rather than an exception) is fine with me.